EncodingStreamingHighAvailability/HighAvailability/Helpers/MediaServicesHelper.cs

```diff
@@ -0,0 +1,172 @@
+            }
+
+            return transform;
+        }
+
+        public static async Task<ContentKeyPolicy> EnsureContentKeyPolicyExists(IAzureMediaServicesClient client, string resourceGroup, string accountName, string contentKeyPolicyName, byte[] tokenSigningKey, string issuer, string audience)
+        {
+            ContentKeyPolicySymmetricTokenKey primaryKey = new ContentKeyPolicySymmetricTokenKey(tokenSigningKey);
+            List<ContentKeyPolicyRestrictionTokenKey> alternateKeys = null;
+            List<ContentKeyPolicyTokenClaim> requiredClaims = new List<ContentKeyPolicyTokenClaim>()
+            {
+                ContentKeyPolicyTokenClaim.ContentKeyIdentifierClaim
+            };
+
+            List<ContentKeyPolicyOption> options = new List<ContentKeyPolicyOption>()
+            {
+                new ContentKeyPolicyOption(
+                    new ContentKeyPolicyClearKeyConfiguration(),
+                    new ContentKeyPolicyTokenRestriction(issuer, audience, primaryKey,
+                        ContentKeyPolicyRestrictionTokenType.Jwt, alternateKeys, requiredClaims))
+            };
+
+            // since we are randomly generating the signing key each time, make sure to create or update the policy each time.
+            // Normally you would use a long lived key so you would just check for the policies existence with Get instead of
+            // ensuring to create or update it each time.
+            ContentKeyPolicy policy = await client.ContentKeyPolicies.CreateOrUpdateAsync(resourceGroup, accountName, contentKeyPolicyName, options).ConfigureAwait(false);
+
+            return policy;
+        }
+
+        public static string GetToken(string issuer, string audience, string keyIdentifier, byte[] tokenVerificationKey)
+        {
+            var tokenSigningKey = new SymmetricSecurityKey(tokenVerificationKey);
+
+            SigningCredentials cred = new SigningCredentials(
+                tokenSigningKey,
+                // Use the  HmacSha256 and not the HmacSha256Signature option, or the token will not work!
+                SecurityAlgorithms.HmacSha256,
+                SecurityAlgorithms.Sha256Digest);
+
+            Claim[] claims = new Claim[]
+            {
+                new Claim(ContentKeyPolicyTokenClaim.ContentKeyIdentifierClaim.ClaimType, keyIdentifier)
+            };
+
+            JwtSecurityToken token = new JwtSecurityToken(
+                issuer: issuer,
+                audience: audience,
+                claims: claims,
+                notBefore: DateTime.Now.AddMinutes(-5),
+                expires: DateTime.Now.AddMinutes(60),
+                signingCredentials: cred);
+
+            JwtSecurityTokenHandler handler = new JwtSecurityTokenHandler();
+
+            return handler.WriteToken(token);
+        }
+
+        public static bool IsSystemError(Job job, string jobOutputAssetName)
```

Comment:
Is this a correct way to determine if job should be resubmitted?

Reply:
Is there not some way to check if the error is due to bad file or User error?

Reply:
Beverly suggested to also check Transform output OnError flag. Added that to job resubmit logic.